GVFS/GVFS/CommandLine/VerbConstants.POSIX.cs

```diff
@@ -0,0 +1,10 @@
+namespace GVFS.CommandLine
+{
+    public static class VerbConstants
```

Comment:
Is there anyway we can add this to the GVFSPlatform instead since that it where differences in the platforms are?  I would hate to start having `.POSIX.cs`, `.Mac.cs`, `.Unix.cs`, and `.Windows.cs` files all over the code base.

Reply:
> Is there anyway we can add this to the GVFSPlatform instead since that it where differences in the platforms are?

I don't think this is possible because we need this for an attribute that is set at compile time:

```
[Option(
  "local-cache-path",
   Required = false,
   HelpText = VerbConstants.CloneVerb.LocalCacheRootHelpText)]
```

The only alternative that comes to mind is that rather than adding the sources files in this folder I could add them in the `GVFS.Platform.Windows` and `GVFS.Platform.POSIX` folders, but the csprojs here would need to link the files in directly (similar to how GVFS.Hooks links in shared files).

Do you think that would be cleaner?

@nickgra, do you have any suggestions for cleaning this up?

Reply:
I just seems to me like we are ending up with multiple ways to have platform specific code but maybe that is just me and there is some standard convention we are following?

Reply:
The general approach we've been using:

- Compile time dependency (e.g. this PR) -> Include platform specific code in csproj
- Project cannot consume GVFS.Common assembly (e.g. GVFS.Hooks) -> Include platform specific code in csproj
- Everything else -> Use GVFSPlatform

@nickgra please double check me here.

I think it's a little more gray in the service work that @alameenshah is working on.  Based on the approach described above we probably should be using GVFSPlatform there for the platform-specific code.

Reply:
I tried some things locally which didn't work out so I don't have any ideas or suggestions on ways to clean this up.

I agree with the theme behind Kevin's comment, things are getting complicated with multiple ways to do platform specific code. I believe this is the first PR in which we've implemented a compile time dependency like this (FastFetch does something similar, but it's a different program), but I don't think we have an option with attributes. Maybe there's another way to specify arguments for CommandLineParser that doesn't use attributes?

Long term, I think we want to see the Hooks go away so that would remove that method.

Reply:
Another (cheap) option would be to put both the default for Windows and the default for POSIX (macOS and Linux) in a string that is shared across both platforms. I don't think that's too poor of a user experience.

Reply:
> Another (cheap) option would be to put both the default for Windows and the default for POSIX (macOS and Linux) in a string that is shared across both platforms. I don't think that's too poor of a user experience.

I didn't initially go with that option because of UX concerns.

@jrbriggs what are your thoughts on this option?

Reply:
\+ @kewillford for his thoughts on this option as well

Reply:
A crazy idea I had was `<DefineConstants>` in the csproj per-platform, and then just ifdef the attribute. Not sure that's much better though :)

Reply:
Yeah I had considered the `#ifdef` route as well, but as a general rule we've been trying to avoid adding `#ifdef` ... looks like there is no perfect option here 🙁

Reply:
@jrbriggs do you if is possible to extract the value of project constants in the source code?

If that was the case then we could add these strings to the csprojs themselves and not need to add new source files for them.

Reply:
C# doesn't support preprocessor-like operations. While it can check whether or not a token was defined, it can't do any replacements like it's C bretheren.

Reply:
Yeah looks like there isn't really a good option.  If the `CommandLineParser` supported localization of the HelpText we could use that but it doesn't.  How important is showing the default location is the question because without that we could have the same text for both platforms.

Reply:
> How important is showing the default location is the question because without that we could have the same text for both platforms.

That's a good point, @kewillford, @jrbriggs, @nickgra what would you think if I just shortened the help text to:

> "Use this option to override the default path for the local GVFS cache."

I doubt that users have been relying on the help text to know where the default location is.

Reply:
> "Use this option to override the default path for the local GVFS cache."

I like this suggestion!

Reply:
I like it.

Reply:
Sounds good, thanks everyone! I will clean this up.